core/rawdb/accessors_chain.go

```diff
@@ -87,6 +89,25 @@ func TruncateCanonicalHash(tx kv.RwTx, blockFrom uint64, markChainAsBad bool) er
 	return nil
 }
 
+func GetLatestBadBlocks(tx kv.Tx, limit int) ([]*types.Block, error) {
+	// BadHeaderNumber table is typically small, so that we can go through all of it.
+	bheap := utils.NewBlockMaxHeap(limit)
```

Comment:
Probably you don't need `heap`:
- table sorted by key. key starts from blockNum.
- can read table backward - 2 ways: `Cursor` has `.Last()/.Prev()` methods. Also `tx` has `.Range` method with `order.Desc` parameter. Both ways allow read `N` keys or until some prefix.

Reply:
the key is hash...so it can't be fetched by ordered values (blockNum)?

Reply:
Ah. Ok.